### ISSUE TITLE
Error when not at git root directory

### ISSUE THREAD

**Issue by hoschi** [PAntoine/vimgitlog#3]
I get an error when open git log for a file when the current dir is a subdirectory of the git root dir:

```
Fehler beim Ausführen von "function GITLOG_ToggleWindows..GITLOG_GetHistory..<SNR>106_GITLOG_OpenTreeWindow":
Zeile   49:
E716: Schlüssel nicht vorhanden im Wörterbuch: lnum)
E116: Ungültige Argumente für die Funktion GITLOG_RedrawTreeWindow
File feature/8:src/service.js is not tracked /home/hoschi/repos/sweetp-code/services/project-context-manager/.git rev HEAD:service.js
```

When I exit vim, I get this as addition

```
fatal: Not a valid object name HEAD:service.js
```

You can try this when opening `src/service.js` vs navigating to the `src` dir and open `service.js` in this repo https://github.com/hoschi/sweetp-service-project-context-manager

**Comment by PAntoine** [comment]
can you tell me what version. 4.0.1?

**Comment by PAntoine** [comment]
Actually, don't worry I have fixed that bug in a newer version I am working on.
Will update when not at the day job.

If you want to do a quick runing fix change line 882 in the plugin/gitlog.vim to:

```
if !has_key(found_item, "lnum")
    let found_item.lnum = 1
endif

call GITLOG_RedrawTreeWindow(found_item.lnum)
```

**Comment by hoschi** [comment]
oh sorry, yes the latest stable version, 4.0.1. Great you fixed that already, I check it out when the new version is ready ;)

**Comment by PAntoine** [comment]
should be fixed @ verison 4.0.2

There are a couple of other bugs but they are either old or should not cause problems. Will be releasing a new version in a few days so did not want to back port the fixes.

I hope it works now.

**Comment by hoschi** [comment]
> Will be releasing a new version in a few days

you mean a new major version with more bug fixes?

**Comment by PAntoine** [comment]
Speed inprovements for big trees. I have re-written the way I build directory trees and trying to reduce the amount of file checking I need to do for the full_tree_walk. currectly the kernel tree (that has been built) takes about 18mins from start. I am trying to get that down to something sensible.

So it will be a minor version, with a couple of bug fixes that I found when testing. It's a couple of days work, when I can get the time. There are no crashy bugs in it (unless I have added some new ones).
